File: 01. Leetcode/faangdsa/sachin/arrays/01/removeDuplicates.ts
function removeDuplicates(nums: number[]): number {
    let l = 1;

    for (let r=1; r<nums.length; r++){
        if (nums[r-1] != nums[r]){
            nums[l] = nums[r];
            l++;
        }
    }
    return l;
};

File: 01. Leetcode/faangdsa/sachin/arrays/01/RemoveDuplicates.scala
object Solution {
    def removeDuplicates(nums: Array[Int]): Int = {
        var l: Int = 1;

        for (r <- 1 until nums.length){
            if (nums(r-1) != nums(r)) {
                nums(l) = nums(r);
                l += 1;
            }
        }

        l;
    }
}